galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.common/src/test/java/dev/galasa/zos3270/common/screens/images/TerminalImageTransformTest.java

```diff
@@ -0,0 +1,311 @@
+        transform.writeImage(image, "png", buff);
+
+        byte[] contentsRendered = buff.toByteArray();
+
+        assertThat(contentsRendered).isNotEmpty();
+
+        String testFileToCompareAgainst = testName.getMethodName()+".png";
+
+        assertFileContentsSame( contentsRendered , testFileToCompareAgainst);
+    }
+
+    private void assertFileContentsSame(byte[] contentsRendered, String testFileToCompareAgainst) throws IOException {
+
+        try ( InputStream testImageToCompare = this.getClass().getClassLoader().getResourceAsStream(testFileToCompareAgainst)
+        ) {
+
+            if (testImageToCompare == null) {
+                String tempFilePath = writeTempFile(testFileToCompareAgainst,contentsRendered);
+                fail("Testcase logic failure. Tried to open file "+testFileToCompareAgainst+" but it was missing."+
+                    "... was hoping to compare it to image\n"+tempFilePath);
+            }
+
+            byte[] goodFileContents = testImageToCompare.readAllBytes();
+
+            boolean isSame = true ;
+
+            if (contentsRendered.length != goodFileContents.length ) {
+                String tempFilePath = writeTempFile(testFileToCompareAgainst,contentsRendered);
+                fail("rendered image is different size to expected."+
+                " ... image from the renderer is here:\n"+tempFilePath);
+            }
+            
+            for( int i=0; isSame && i<contentsRendered.length; i++) {
+                byte got = contentsRendered[i];
+                byte expected = goodFileContents[i];
+                if (got != expected) {
+                    isSame = false;
+                }
+            }
+
+            if (!isSame) {
+                String tempFilePath = writeTempFile(testFileToCompareAgainst,contentsRendered);
+                fail("The rendered image is not the same as the one stored at "+"image-data/"+testFileToCompareAgainst+
+                " ... got this image from the renderer:\n"+tempFilePath);
+            }
+        }
+    }
+
+    private String writeTempFile(String name, byte[] bytes) throws IOException {
+        java.nio.file.Path tempDir = Files.createTempDirectory("TerminalImageRendererTest");
+        java.nio.file.Path filePath = tempDir.resolve(name);
+
+        try ( OutputStream outStream = Files.newOutputStream(filePath, StandardOpenOption.CREATE); 
+        ) {
+            outStream.write(bytes);
+        }
+        System.out.println("Image file "+filePath.toAbsolutePath()+" created.");
+
+        return filePath.toAbsolutePath().toString();
+    }
+
+    private Character[] stringToCharacterArray(String text ) {
+        Character[] result = new Character[text.length()];
+
+        char[] charArray = text.toCharArray();
+        for( int i=0; i<charArray.length; i++ ) {
+            result[i] = new Character( charArray[i]);
+        }
+        return result;
+    }
+
+    @Test
+    public void testTextFieldRendersOk() throws Exception {
```

Comment:
Add a test of a field at (0,0) to make sure the offsets are ok.
Add a test which is 80 chars long in a row which should be able to fit 80 chars.

Reply:
Both tests added, plus one which uses all rows.